### PR TITLE
[7.x] Add Publish State permission

### PR DESCRIPTION
**Pull request by edalzell**
Matching the `canManagePublishState` from core and replicating this PR: https://github.com/statamic/cms/pull/10424

### CODE REVIEW

**Comment by edalzell** [comment]
@duncanmcclean let's talk about this one, as I believe we'd need a migration script otherwise it's breaking.

**Comment by duncanmcclean** [comment]
> let's talk about this one, as I believe we'd need a migration script otherwise it's breaking.

Yes, we'd either need to ship with an upgrade script or include this PR in the next breaking changes release (which'll have the nested fields fix).

File: src/ServiceProvider.php [hunk 1]
```diff
@@ -149,6 +149,9 @@ protected function registerPermissions(): self
                                     Permission::make("create {$resource->handle()}")
                                         ->label($this->permissionLabel('create', $resource)),
 
+                                    Permission::make("publish {$resource->handle()}")
```
Comment (duncanmcclean) [inline src/ServiceProvider.php:152]: The concept of "publish states" are opt-in, so we probably only want to add this permission if they're enabled for the resource.

```php
if ($resource->hasPublishStates()) {
    Permission::make("publish {$resource->handle()}")
        ->label("Manage {$resource->handle()} Publish State"),
}
```

File: resources/js/components/Publish/PublishForm.vue [hunk 2]
```diff
@@ -158,8 +158,8 @@
                     class="rtl:mr-4 ltr:ml-4 btn-primary flex items-center"
                     :disabled="!canPublish"
                     @click="confirmingPublish = true">
-                    <span v-text="__('Publish')" />
-                    <svg-icon name="micro/chevron-down-xs" class="rtl:mr-2 ltr:ml-2 w-2" />
+                    <span v-text="this.canManagePublishState ? __('Publish') : __('Create Revision')" />
```
Comment (duncanmcclean) [inline resources/js/components/Publish/PublishForm.vue:161]: Like we've done in Core, can we extract this out into a `publishButtonText` method?

File: resources/js/components/Publish/PublishForm.vue
Comment (duncanmcclean) [inline resources/js/components/Publish/PublishForm.vue:1]: Related to [my comment here](https://github.com/statamic/cms/pull/10424#discussion_r1672091778), you updated the text of multiple buttons in this PR, but only updated one button in your Core PR.

Is there any particular reason for this?

**Review by duncanmcclean — CHANGES_REQUESTED** [review]

**Review by duncanmcclean — APPROVED** [review]

**Pull request merged**